From 31b30cafb1267b5cc5bfbf84dea7898784ddae04 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 17 Oct 2017 12:31:16 +0800
Subject: [PATCH 30/35] btwilink: add minimal device tree support

This commit comes from branch ti-linux-3.12.y:
  git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git

Add minimal device tree support to the btwilink driver that is used
for binding bluetooth with the ti-st shared transport driver.

Change-Id: I301c49d29046f20f8868bebb14347e82c12c8140
Signed-off-by: Eyal Reizer <eyalr@ti.com>
Signed-off-by: bvijay <bvijay@ti.com>
(cherry picked from commit 61214ac0fa94777994d8d7076e53bd4a7ebc2a82)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/bluetooth/btwilink.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/bluetooth/btwilink.c b/drivers/bluetooth/btwilink.c
index b6bb58c..e6db946 100644
--- a/drivers/bluetooth/btwilink.c
+++ b/drivers/bluetooth/btwilink.c
@@ -30,6 +30,7 @@
 
 #include <linux/ti_wilink_st.h>
 #include <linux/module.h>
+#include <linux/of.h>
 
 /* Bluetooth Driver Version */
 #define VERSION               "1.0"
@@ -275,6 +276,14 @@ static int ti_st_send_frame(struct hci_dev *hdev, struct sk_buff *skb)
 	return 0;
 }
 
+static const struct of_device_id btwilink_of_match[] = {
+{
+	.compatible = "btwilink",
+	},
+	{}
+};
+MODULE_DEVICE_TABLE(of, btwilink_of_match);
+
 static int bt_ti_probe(struct platform_device *pdev)
 {
 	static struct ti_st *hst;
@@ -338,6 +347,7 @@ static int bt_ti_remove(struct platform_device *pdev)
 	.remove = bt_ti_remove,
 	.driver = {
 		.name = "btwilink",
+		.of_match_table = of_match_ptr(btwilink_of_match),
 	},
 };
 
-- 
1.7.9.5

